### PR TITLE
fix #4067: put jobs in modal

### LINKED ISSUE #4067: jobs task as a modal window
The Jobs link in the user menu takes you out of the app an to a separate page. Instead it should show a modal dialog within the app with "Refresh" and "Close" buttons.

### PR DESCRIPTION
**Pull request by rorour**

### CODE REVIEW

File: sirepo/package_data/static/js/sirepo-components.js [hunk 1]
```diff
@@ -880,25 +880,27 @@ SIREPO.app.directive('logoutMenu', function(authState, authService, requestSende
...
-                '<li class="dropdown-header">{{ authState.paymentPlanName() }}</li>',
-                '<li class="dropdown-header" data-ng-if="::authState.userName">{{ ::authState.userName }}</li>',
-                '<li data-ng-if="showAdmJobs()"><a data-ng-href="{{ getUrl(\'admJobs\') }}">Admin</a></li>',
-                '<li><a data-ng-href="{{ getUrl(\'ownJobs\') }}">Jobs</a></li>',
-                '<li><a data-ng-href="{{ ::authService.logoutUrl }}">Sign out</a></li>',
-              '</ul>',
-            '</li>',
-        ].join(''),
-        controller: function($scope) {
+        template: `
```
Comment (e-carlin) [inline sirepo/package_data/static/js/sirepo-components.js:883]: Thanks for converting to new style:)

File: sirepo/package_data/static/js/sirepo-components.js [hunk 2]
```diff
@@ -909,6 +911,11 @@ SIREPO.app.directive('logoutMenu', function(authState, authService, requestSende
             $scope.showAdmJobs = function() {
                 return authState.roles.indexOf('adm') >= 0;
             };
+
+            $scope.showJobsList = function() {
+                $('#' + panelState.modalId('jobsListModal')).prependTo("body");
```
Comment (e-carlin) [inline sirepo/package_data/static/js/sirepo-components.js:916]: If you move the `<div data-jobs-list-modal="" data-title="Jobs" data-id="sr-jobsListModal-editor"></div>` to the commonFooter directive then you shouldn't need this line.

File: sirepo/package_data/static/js/sirepo-components.js [hunk 3]
```diff
@@ -3305,7 +3311,41 @@ SIREPO.app.directive('jobsList', function(requestSender, appState, $location, $s
             appState.clearModels(appState.clone(SIREPO.appDefaultSimulationValues));
             $scope.getJobs();
 
+            panelState.waitForUI(() => {
+                $('#' + panelState.modalId('jobsListModal')).on('shown.bs.modal', function() {
+                $scope.getJobs();
+                });
+            });
+
```
Comment (e-carlin) [inline sirepo/package_data/static/js/sirepo-components.js:3319]: Remove this blank line and the one 2 lines down.

**Review by e-carlin — COMMENTED** [review]
Looks good. Few comments. As we discussed in person also need to:
- remove own-jobs.html
- remove traces of it from schema-common.json
- make it so when you click on the "pending link" (happens when one sim is waiting on another) it also opens in a modal.

File: sirepo/package_data/static/js/sirepo-components.js (hunk 1 above)
Comment (rorour) [reply sirepo/package_data/static/js/sirepo-components.js:883]: 3f4d10d86518e0e428966ed7697717e1d82d6e96

File: sirepo/package_data/static/js/sirepo-components.js (hunk 3 above)
Comment (rorour) [reply sirepo/package_data/static/js/sirepo-components.js]: 3f4d10d86518e0e428966ed7697717e1d82d6e96

**Comment by e-carlin** [comment]
Looks good! One more thing - when you go to the "admin jobs" page now you need to click refresh before the jobs list loads. Can it be made so it loads when you go to the page? A bit hard to describe what I mean so let me know if that doesn't make sense.

File: sirepo/package_data/static/js/sirepo-components.js [hunk 4]
```diff
@@ -3312,7 +3312,7 @@ SIREPO.app.directive('jobsList', function(requestSender, appState, $location, $s
             };
 
             appState.clearModels(appState.clone(SIREPO.appDefaultSimulationValues));
-
```
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/package_data/static/js/sirepo-components.js:3307]: Related to https://github.com/radiasoft/sirepo/issues/4073?

[the remark this answers is not in the document]
Comment (e-carlin) [reply sirepo/package_data/static/js/sirepo-components.js:3307]: No, related to how we put modals in common-footer. The modal is then loaded, we make the request for jobs but we aren't logged in fully, and get kicked back out

[the remark this answers is not in the document]
Comment (e-carlin) [reply sirepo/package_data/static/js/sirepo-components.js:3307]: https://github.com/radiasoft/sirepo/pull/4115/commits/d244cdcce04d2d9564729b7dbb0e70ac131f9862

File: sirepo/package_data/static/js/sirepo-components.js [hunk 5]
```diff
@@ -3303,9 +3312,44 @@ SIREPO.app.directive('jobsList', function(requestSender, appState, $location, $s
             };
 
             appState.clearModels(appState.clone(SIREPO.appDefaultSimulationValues));
-            $scope.getJobs();
+            $scope.$on('$routeChangeSuccess', () => {
+                if ($location.path() == (SIREPO.APP_SCHEMA.route.admJobs)) {
```
Comment (e-carlin) [inline sirepo/package_data/static/js/sirepo-components.js:3316]: @moellep do you have a better way to do this than checking for routeChangeSuccess? The case this is handling is when we go to the adm-jobs page we want the jobs list to be refreshed on first load. W/o this feature the jobs list is empty and we have click the "refresh" button to load the list of jobs.

**Pull request merged**